### PR TITLE
feat(rsa): always use the fastest method available for rsa keygeneration

### LINKED ISSUE #9: Enable WebWorkers to make it go faster
node-forge supports WebWorkers
image

enable here: https://github.com/libp2p/js-libp2p-crypto/blob/master/src/keys/rsa.js#L131

### PR DESCRIPTION
**Pull request by dignifiedquire**
BREAKING CHANGE: generateKeyPair is now async

Simple secio benchmarks of key gen, (node.js)
- before this change: 0.27 ops/sec
- after this change (with ursa): 8.47 ops/sec
- after this change (without ursa): 0.52 ops/sec

Fixes #9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "libp2p-crypto",
   "version": "0.6.1",
   "description": "Crypto primitives for libp2p",
-  "main": "lib/index.js",
+  "main": "src/index.js",
```
Comment (dignifiedquire) [inline package.json]: Todo: revert before merge

File: src/keys/rsa.js [hunk 2]
```diff
@@ -127,9 +127,39 @@ function unmarshalRsaPublicKey (bytes) {
...
 
-function generateKeyPair (bits) {
-  const p = rsa.generateKeyPair({bits})
-  return new RsaPrivateKey(p.privateKey, p.publicKey)
+function fastKeys (bits, cb) {
+  if (typeof window === 'undefined') {
+    let ursa
+    try {
+      ursa = require('ursa')
+    } catch (err) {
```
Comment (VictorBjelkholm) [inline src/keys/rsa.js]: Should at least show a warning if something goes wrong here

Comment (dignifiedquire) [inline src/keys/rsa.js]: Why? This is simply the check if the optional dependency is available to be used

Comment (VictorBjelkholm) [inline src/keys/rsa.js]: Ah, did not know is what an optional dependency and my spidey senses goes off when I see a empty catch in a try/catch block.

Comment (dignifiedquire) [inline src/keys/rsa.js]: As they should :)

**Comment by dignifiedquire** [comment]
Closing in favor of #12

**Pull request closed without merging**